### ISSUE TITLE
Disable/Remove mako

### ISSUE THREAD

**Issue by carlosV2** [manjaro-sway/manjaro-sway#330]
Hi,

Some time ago I replaced `mako` with [`SwayNotificationCenter`](https://github.com/ErikReider/SwayNotificationCenter). The way I did it was to create the file `~/.config/sway/definitions.d/40-notifications.conf` with the following content:
```
# set $notifications mako -c $HOME/.config/mako/config
set $notifications swaync
```

It worked for a while but I recently updated SwayNotificationCenter and it stopped working (it runs mako after every boot).
If I `kill -9 <mako's pid>` and then run `swaync`, the notifications work exactly as expected but I'm unable to get it working after boot.

What can I try?

**Comment by boredland** [comment]
I think what you do is exactly right. 
[recording_20220922-104015.webm](https://user-images.githubusercontent.com/4662748/191700415-8b06ce6e-8170-418a-bc8c-4f84a443cc9e.webm)

**Comment by carlosV2** [comment]
Hi @boredland 😄, thanks for stopping by.

This was my first thought too but I couldn't find any:
```
$ grep -rnw /etc/sway/ -e mako
/etc/sway/definitions:69:set $notifications /usr/share/sway/scripts/mako.sh --font "$term-font" --text-color "$text-color" --border-color "$accent-color" --background-color "$background-color" --border-size 3 --width 400 --height 200 --padding 20 --margin 20 --default-timeout 15000

$ grep -rnw ~/.config/sway/ -e mako
<HOME>/.config/sway/definitions.d/40-notifications.conf:1:# set $notifications mako -c $HOME/.config/mako/config
```

I also wanted to try the hard way and remove `mako` completely off my system but it is required by `manjaro-sway-settings` so I though on not removing it for now unless there is no other option. 😞

What I don't get is how is `mako` even being run? Even if `swaync` command failed misserably, what is pulling `mako` up?

**Comment by boredland** [comment]
depending on how old your config is, there may be an issue in .config/sway/config - you could diff that with the version in /etc/skel/.config/sway/config.

**Comment by carlosV2** [comment]
Good point! Still no luck tho 😞 
```
$ md5sum ~/.config/sway/config
13aa86f26eaaa7347a4bce9e2d12f3f8  <HOME>/.config/sway/config

$ md5sum /etc/skel/.config/sway/config
13aa86f26eaaa7347a4bce9e2d12f3f8  /etc/skel/.config/sway/config
```

For record, this is the content of those files:
```
# Default config for sway
#
# Copy this to ~/.config/sway/config and edit it to your liking.
#
# Read `man 5 sway` for a complete reference.

# styles - set the folder for your theme definition file
set $theme /usr/share/sway/themes/matcha-green

# theme variables
include $theme/definitions

# global variables
include /etc/sway/definitions

# user variables
include $HOME/.config/sway/definitions.d/*.conf

include /etc/sway/inputs/*
include /etc/sway/outputs/*

# enable modes
include /etc/sway/modes/*

# only enable this if every app you use is compatible with wayland
# xwayland disable

# include additional configs e.g. to autostart applications
include /etc/sway/config.d/*

# user config
include $HOME/.config/sway/config.d/*.conf
```

I've been checking this file and the order of definition/execution seems correct. It also seems like it is redeclaring the variable in the right order.

I'm really puzzled with this behaviour... 🤔 

I'm tempted to think that maybe the `~/.config/sway/` files aren't being loaded but I do have other configurations in there and they are picked up correctly.

**Comment by boredland** [comment]
Hard to find out, I guess. Is there perhaps anything calling it from `config/sway/config.d` ?

**Comment by carlosV2** [comment]
And just like that, it got solved... 🤷🏼‍♂️ 

Another update came today plus a bunch of other packages and it got resolved alone. I still don't know what happened or why `mako` was being executed. My only assumption is that a bug or something, run `mako` before `swaync` had a chance to run and since those are compeating software, `swaync` realised there was another notification daemon running and shut itself off.

So, in conclusion, the configuration was correct and `sway` was, apparently, following the configuration but something got `mako` invoked which prevented `swaync` to run.

![recording](https://user-images.githubusercontent.com/2642839/192021796-ce509492-c179-4306-9537-8e33577f4f55.gif)

**Comment by carlosV2** [comment]
Ok, it happened again... For whatever reason, mako is running again.

I've remembered the `pstree` command and, as expected it is running and it seems to be running off `systemd`:
```
$ pstree
systemd-+-NetworkManager---2*[{NetworkManager}]
        |-systemd-+-(sd-pam)
        |         |-mako
```

However, I can't find any obvious service that runs it:
```
$ systemctl list-units --type=service
  UNIT                                                LOAD   ACTIVE SUB     DESCRIPTION
  apparmor.service                                    loaded active exited  Load AppArmor profiles
  auto-cpufreq.service                                loaded active running auto-cpufreq - Automatic CPU speed & power optimizer for Linux
  avahi-daemon.service                                loaded active running Avahi mDNS/DNS-SD Stack
  bluetooth.service                                   loaded active running Bluetooth service
  cups.service                                        loaded active running CUPS Scheduler
  dbus.service                                        loaded active running D-Bus System Message Bus
  docker.service                                      loaded active running Docker Application Container Engine
  greetd.service                                      loaded active running Greeter daemon
  kmod-static-nodes.service                           loaded active exited  Create List of Static Device Nodes
  lvm2-monitor.service                                loaded active exited  Monitoring of LVM2 mirrors, snapshots etc. using dmeventd or progress polling
  NetworkManager-wait-online.service                  loaded active exited  Network Manager Wait Online
  NetworkManager.service                              loaded active running Network Manager
  polkit.service                                      loaded active running Authorization Manager
  rtkit-daemon.service                                loaded active running RealtimeKit Scheduling Policy Service
  snapd.apparmor.service                              loaded active exited  Load AppArmor profiles managed internally by snapd
  snapd.service                                       loaded active running Snap Daemon
  systemd-backlight@backlight:intel_backlight.service loaded active exited  Load/Save Screen Backlight Brightness of backlight:intel_backlight
  systemd-backlight@leds:dell::kbd_backlight.service  loaded active exited  Load/Save Screen Backlight Brightness of leds:dell::kbd_backlight
  systemd-binfmt.service                              loaded active exited  Set Up Additional Binary Formats
  systemd-journal-flush.service                       loaded active exited  Flush Journal to Persistent Storage
  systemd-journald.service                            loaded active running Journal Service
  systemd-logind.service                              loaded active running User Login Management
  systemd-modules-load.service                        loaded active exited  Load Kernel Modules
  systemd-random-seed.service                         loaded active exited  Load/Save Random Seed
  systemd-remount-fs.service                          loaded active exited  Remount Root and Kernel File Systems
  systemd-sysctl.service                              loaded active exited  Apply Kernel Variables
  systemd-tmpfiles-setup-dev.service                  loaded active exited  Create Static Device Nodes in /dev
  systemd-tmpfiles-setup.service                      loaded active exited  Create Volatile Files and Directories
  systemd-udev-trigger.service                        loaded active exited  Coldplug All udev Devices
  systemd-udevd.service                               loaded active running Rule-based Manager for Device Events and Files
  systemd-update-utmp.service                         loaded active exited  Record System Boot/Shutdown in UTMP
  systemd-user-sessions.service                       loaded active exited  Permit User Sessions
  tlp.service                                         loaded active exited  TLP system startup/shutdown
  udisks2.service                                     loaded active running Disk Manager
  upower.service                                      loaded active running Daemon for power management
  user-runtime-dir@1000.service                       loaded active exited  User Runtime Directory /run/user/1000
  user@1000.service                                   loaded active running User Manager for UID 1000
  wpa_supplicant.service                              loaded active running WPA supplicant

LOAD   = Reflects whether the unit definition was properly loaded.
ACTIVE = The high-level unit activation state, i.e. generalization of SUB.
SUB    = The low-level unit activation state, values depend on unit type.
38 loaded units listed. Pass --all to see loaded but inactive units, too.
To show all installed unit files use 'systemctl list-unit-files'.
```

Any idea how can I check what service is triggering it?

**Comment by Mathew-D** [comment]
what happens if you edit the /etc/sway/definitions file directly and comment out the 
set $notifications /usr/share/sway/scripts/mako.sh line
as I am petty sure that is the only thing that starts mako for us.

**Comment by carlosV2** [comment]
I'll try it and report it back but, if this is true, it means there is something very wrong in the way sway loads the configuration files 😞.  From what I can tell, any other config file is loaded correctly but this one 🤔.

**Comment by Mathew-D** [comment]
Well I just installed swaync and 
int my ~/.config/sway/definitions.d/00-user-define.conf  
put 
set $notifications swaync
rebooted and it started with no mako

**Comment by carlosV2** [comment]
@Mathew-D I'm afraid it didn't work.

I updated `/etc/sway/definitions` as per your suggestion leaving it as follows:
```
$ cat /etc/sway/definitions
...

###Notification daemon configuration
# set $notifications /usr/share/sway/scripts/mako.sh --font "$term-font" --text-color "$text-color" --border-color "$accent-color" --background-color "$background-color" --border-size 3 --width 400 --height 200 --padding 20 --margin 20 --default-timeout 15000
set $notifications swaync

...
```

It did not make any difference. After a full restart, `mako` is again the running notifications daemon 😩.

**Comment by Mathew-D** [comment]
well the shit answer is to
` sudo mv /usr/bin/mako /usr/bin/mako-b`
I am not sure what could be running mako.
When  you ps ux is mako near the top that might give you some idea when it is starting.

**Comment by Mathew-D** [comment]
I think I found your problem.
"NetworkManager-wait-online.service blocks until NetworkManager logs "startup complete" and announces startup complete on D-Bus."
That service is trying to announce which I think means it is trying to use mako to do it.  Try turning that service off and see if that fixes your problem.

**Comment by varac** [comment]
Fyi, I stumbled here because I had the same issue that mako magically was started on boot even though I overrode the $mako and $notifications definition like this:

```
$ cat ~/.config/sway/definitions.d/custom.conf
set $notifications swaync
set $mako swaync
```

I noticed this line in the journal:

```
 dbus-daemon[977]: [session uid=1000 pid=977] Activating via systemd: service name='org.freedesktop.Notifications' unit='mako.service' requested by ':1.34' (uid=1000 pid=1161 comm="poweralertd -s -i line power")
```

And found that a systemd *user* session was still active (using the `/usr/lib/systemd/user/mako.service` service definition).

I disabled it by `systemctl --user mask mako.service` and now it doesn't start anymore.

**Comment by carlosV2** [comment]
Yes! YEEEEEEES!

It worked! I almost had an orgasm when I finally saw swaync starting on it's own after all this time!

It turns out I was not only missing the `systemctl` call but also the `set $mako swaync` part!
A million thanks for sharing it. You are a true hero!

Definitely not all heroes wear capes!

**Comment by carlosV2** [comment]
Yes! YEEEEEEES!

It worked! I almost had an orgasm when I finally saw swaync starting on it's own after all this time!

It turns out I was not only missing the `systemctl` call but also the `set $mako swaync` part!
A million thanks for sharing it. @varac you are a true hero!

Definitely not all heroes wear capes!

**Comment by boredland** [comment]
Very nice! We should probably rename `$mako` to something less vendor-specific and add this to our support.md - In general I'd be open to a PR switching out mako with another default. I have not strong opinions on that and never missed anything with mako.

**Comment by varac** [comment]
@carlosV2 Nice that it works for you, you're welcome!
